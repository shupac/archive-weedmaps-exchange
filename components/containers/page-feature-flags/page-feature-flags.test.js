import React, { Component } from 'react';
import urlConfig from 'lib/common/url-config';
import { mount } from 'enzyme';
import toJSON from 'enzyme-to-json';
import nock from 'nock';
import getFeatureFlags from './';

class MockPage extends Component {
  static getInitialProps() {}

  render() {
    return <div>Mock Page</div>;
  }
}

describe('getFeatureFlags', () => {
  it('should be able to wrap a page component with a FeatureFlagProvider', () => {
    const WrappedMockPage = getFeatureFlags(
      MockPage,
      {
        baseUrl: urlConfig.featureFlagApiUrl,
      },
      ['test-feature-flag'],
    );

    expect(toJSON(mount(<WrappedMockPage />))).toMatchSnapshot();
  });

  it('should be able to get the initial flags', async () => {
    nock(urlConfig.featureFlagApiUrl)
      .head('/test-feature-flag')
      .reply(204, { status: 204 });

    const WrappedMockPage = getFeatureFlags(
      MockPage,
      {
        baseUrl: urlConfig.featureFlagApiUrl,
      },
      ['test-feature-flag'],
    );

    const props = await WrappedMockPage.getInitialProps({
      isServer: true,
    });

    expect(props).toMatchObject({
      initialFlags: {
        'test-feature-flag': true,
      },
    });
  });
});
