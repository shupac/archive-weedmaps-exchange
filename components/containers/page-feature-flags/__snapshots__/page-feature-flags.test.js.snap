// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getFeatureFlags should be able to wrap a page component with a FeatureFlagProvider 1`] = `
<PageWithFeatureFlags>
  <e
    baseUrl="https://api-g.weedmaps.com/feature-flag/v1"
  >
    <MockPage>
      <div>
        Mock Page
      </div>
    </MockPage>
  </e>
</PageWithFeatureFlags>
`;
